scripts/fix_db/current.py

```diff
@@ -11,92 +6,217 @@
+                "replace with project timestamp."
             )
-            stm = select(ApplyWorkflow).where(
-                ApplyWorkflow.project_id == project.id
+            project = db.get(Project, workflow.project_id)
+            if project is None:
+                raise IntegrityError(
+                    f"[Workflow {workflow.id:4d}] "
+                    f"project_id={workflow.project_id}, "
+                    f"but Project {workflow.project_id} does not exist"
+                )
+            new_timestamp = project.timestamp_created
+            logging.warning(
+                f"[Workflow {workflow.id:4d}] New value: {new_timestamp=}"
             )
-            jobs = db.execute(stm).scalars().all()
-            if len(jobs) == 0:
-                logging.warning(
-                    f"[Project {project.id:4d}] No jobs found, skip."
+            workflow.timestamp_created = new_timestamp
+            db.add(workflow)
+            db.commit()
+            db.refresh(workflow)
+            db.expunge(workflow)
+            WorkflowRead(
+                **workflow.model_dump(),
+                task_list=workflow.task_list,
+                project=workflow.project,
+            )
+
+    # add timestamp_created to Dataset
+    stm = select(Dataset)
+    datasets = db.execute(stm).scalars().all()
+    for dataset in datasets:
+        # add timestamp_created to Datasets
+        timestamp_created = dataset.timestamp_created
+        if timestamp_created != REFERENCE_TIMESTAMP:
+            logging.warning(
+                f"[Dataset {dataset.id:4d}] {timestamp_created=} -> skip."
+            )
+        else:
+            logging.warning(
+                f"[Dataset {dataset.id:4d}] {timestamp_created=} -> "
+                "replace with project timestamp."
+            )
+            project = db.get(Project, dataset.project_id)
+            if project is None:
+                raise IntegrityError(
+                    f"[Dataset {dataset.id:4d}] "
+                    f"project_id={dataset.project_id}, "
+                    f"but Project {dataset.project_id} does not exist"
                 )
-                continue
-            timestamp_created = min([job.start_timestamp for job in jobs])
+            new_timestamp = project.timestamp_created
             logging.warning(
-                f"[Project {project.id:4d}] New value: {timestamp_created=}"
+                f"[Dataset {dataset.id:4d}] New value: {new_timestamp=}"
             )
-            project.timestamp_created = timestamp_created
-            db.add(project)
+            dataset.timestamp_created = new_timestamp
+            db.add(dataset)
             db.commit()
-            db.refresh(project)
-            db.expunge(project)
-            ProjectRead(**project.model_dump())
+            db.refresh(dataset)
+            db.expunge(dataset)
+            DatasetRead(
+                **dataset.model_dump(),
```

Comment:
I suggest we explicitly exclude the attributes which are then included manually, for clarity.